link/uprobe.go

```diff
@@ -34,6 +35,9 @@ type Executable struct {
 	path string
 	// Parsed ELF and dynamic symbols' addresses.
 	addresses map[string]uint64
```

Comment:
Maybe we can simplify the population with [`sync.Once`](https://pkg.go.dev/sync#Once.Do)  here instead of introducing an additional `bool` and `sync.Mutex`?